(W.I.P) Lint import aliases using go-import-alias

Glyphack opened the pull request:
FIxes #3021
Used https://github.com/Glyphack/go-import-alias to fix inconsistent import alias names according to `CONTRIBUTING.MD`
Signed-off-by: Shayegan Hooshyari <sh.hooshyari@gmail.com>

stevesloka commented:
Hey @Glyphack, sorry but we just removed all the v2 xDS resource api's from Contour and this broken your entire PR. Also I don't see your .golangci-lint config changes. It would be neat to have those on a commit, then on a second commit changes to make sure the imports are proper. 

Also we should agree on how the imports are going to be aliased. There are some that are super log in the go-control-plane project. 

Thanks!

Glyphack commented:
> Hey @Glyphack, sorry but we just removed all the v2 xDS resource api's from Contour and this broken your entire PR. Also I don't see your .golangci-lint config changes. It would be neat to have those on a commit, then on a second commit changes to make sure the imports are proper.
I tried creating a custom linter for https://github.com/projectcontour/contour/issues/3021, since we haven't talk about how to integrate the linter (I guess there are two options: go vet and golangci-lint) I didn't touch the .golangci-lint. I Think I can fix it when we make a decision.

> Also we should agree on how the imports are going to be aliased. There are some that are super log in the go-control-plane project.
Yes me and @jpeach talked about a rule and I implemented that. you can see the rule [here](https://github.com/projectcontour/contour/issues/3021#issuecomment-711387408)

Finally I implemented a `--fix` option for linter and used it on contour main branch to see it's result(this PR), You and others may want to take a look.
